### PR TITLE
add update&delete apis

### PR DESCRIPTION
**Pull request by mahdiKhd**

### CODE REVIEW

File: news/api/urls.py [hunk 1]
```diff
@@ -4,11 +4,14 @@
...
     path('', views.LandingPageView.as_view(), name='index'),
-    path('<str:token>/', views.NewsDetailView.as_view()),
     path('sign_up/', views.user_signup),
     path('login/', views.login),
     path('categories/<str:category_name>/', views.CategoryDetailView.as_view(), name='category_name'),
     path('add/', views.AddNews.as_view()),
     path('update/<str:token>', views.UpdateNewsView.as_view()),
+    path('delete/<str:token>', views.DeleteNewsView.as_view()),
+    path('createAgency', views.CreateAgencyView.as_view()),
+    path('addReporter', views.AddReporterView.as_view()),
```
Comment (kysre) [inline news/api/urls.py]: Use snake_case instead of camelCase

File: news/services/news.py [hunk 2]
```diff
@@ -97,6 +110,90 @@ def create_news_service(data, reporter):
...
+        setattr(news, field, value)
+    news.save()
+
+    # save changes into cache
+    resp = NewsSerializer(news).data
+    NewsCache.set_news_data(news.token, resp)
+    return True, 'news updated successfully'
+
+
+def delete_news(news):
```
Comment (kysre) [inline news/services/news.py:136]: you should check that the reporter is the author of the news and then delete it

File: news/services/news.py [hunk 3]
```diff
@@ -97,6 +110,90 @@ def create_news_service(data, reporter):
...
+    news.delete()
+
+    # todo delete from cache
+    return True, 'news deleted successfully'
+
+
+def create_agency(data, username):
+    # check data to include require fields and do not include any other fields
+    require_fields = ['name', 'description']
```
Comment (kysre) [inline news/services/news.py]: typo required_fields*

File: news/services/news.py [hunk 4]
```diff
@@ -97,6 +110,90 @@ def create_news_service(data, reporter):
...
+    if not all(key in require_fields for key in data):
+        return False, 'Bad data format'
+
+    # check if there is any other agency with this name
+    agency = get_agency(data['name'])
+    if agency is not None:
+        return False, 'agency with this name already exists'
+
+    # create agency
+    agency = Agency(**data)
```
Comment (kysre) [inline news/services/news.py]: Are you sure that this works?

File: news/services/news.py [hunk 5]
```diff
@@ -97,6 +110,90 @@ def create_news_service(data, reporter):
...
+    reporter.save()
+    # todo save in cache?
+    return True, 'Agency created successfully'
+
+
+def add_reporter(data, reporter):
+    # check data to include require fields and do not include any other fields
+    if 'username' not in data.keys():
+        return False, 'data must include username'
+    if len(data) > 1:
```
Comment (kysre) [inline news/services/news.py]: I think this is unnecessary

File: news/utils/news.py [hunk 6]
```diff
@@ -29,6 +29,22 @@ def get_reporter(username):
         return None
 
 
+def get_subscriber(username):
```
Comment (kysre) [inline news/utils/news.py]: I think get_subscriber_by_username is a better name for this function

File: news/utils/news.py [hunk 7]
```diff
@@ -29,6 +29,22 @@ def get_reporter(username):
...
+    try:
+        subscriber = get_user_model().objects.get(username=username).subscriber
+        return subscriber
+    except get_user_model().DoesNotExist:
+        return None
+
+
+def get_agency(name):
```
Comment (kysre) [inline news/utils/news.py]: get_agency_by_name*

File: news/services/news.py [hunk 8]
```diff
@@ -97,6 +110,90 @@ def create_news_service(data, reporter):
...
+        return False, 'bad data format'
+
+    # check if there is a subscriber with username
+    username = data['username']
+    subscriber = get_subscriber(username)
+    if subscriber is None:
+        return False, 'there is no user with this username'
+
+    # check if there is a reporter with username
+    user_reporter = get_reporter(username)
```
Comment (kysre) [inline news/services/news.py]: get_reporter_by_username

File: news/services/news.py (hunk 2 above)
Comment (mahdiKhd) [reply news/services/news.py:136]: i check it in views by using is_reporter_author function

File: news/api/urls.py (hunk 1 above)
Comment (mahdiKhd) [reply news/api/urls.py]: done

File: news/services/news.py (hunk 3 above)
Comment (mahdiKhd) [reply news/services/news.py]: done

File: news/utils/news.py (hunk 6 above)
Comment (mahdiKhd) [reply news/utils/news.py]: sure

File: news/utils/news.py (hunk 7 above)
Comment (mahdiKhd) [reply news/utils/news.py]: done

File: news/services/news.py (hunk 8 above)
Comment (mahdiKhd) [reply news/services/news.py]: done

**Pull request merged**